src/net_processing.cpp

```diff
@@ -1880,16 +1886,35 @@ PeerManagerInfo PeerManagerImpl::GetInfo() const
     };
 }
 
-void PeerManagerImpl::AddToCompactExtraTransactions(const CTransactionRef& tx)
+void PeerManagerImpl::AddToCompactExtraTransactions(const CTransactionRef& tx, NodeId sender)
 {
     if (m_opts.max_extra_txs <= 0)
         return;
-    if (!vExtraTxnForCompact.size())
+    if (!vExtraTxnForCompact.size()) {
         vExtraTxnForCompact.resize(m_opts.max_extra_txs);
+        m_extra_txn_senders = std::vector<NodeId>(m_opts.max_extra_txs, -1);
+    }
     vExtraTxnForCompact[vExtraTxnForCompactIt] = tx;
+    m_extra_txn_senders[vExtraTxnForCompactIt] = sender;
     vExtraTxnForCompactIt = (vExtraTxnForCompactIt + 1) % m_opts.max_extra_txs;
 }
 
+std::optional<PeerManagerImpl::PackageToValidate> PeerManagerImpl::Find1P1CPackageExtraTxn(const CTransactionRef& orphan_child,
+                                                                                           NodeId child_sender)
+{
+    for (size_t i = 0; i < vExtraTxnForCompact.size(); i++) {
+        const auto& maybe_parent = vExtraTxnForCompact[i];
+        // We stopped checking it earlier due to fee reasons, try again?
+        if (maybe_parent && m_recent_rejects_reconsiderable.contains(maybe_parent->GetWitnessHash().ToUint256())) {
+            Package maybe_cpfp_package{maybe_parent, orphan_child};
+            if (IsChildWithParents(maybe_cpfp_package) && !m_recent_rejects_reconsiderable.contains(GetPackageHash(maybe_cpfp_package))) {
```

Comment:
Feels inefficient/expensive to construct a package + call these checks for every tx every time there's a missing input. Maybe parameterize by the txid of the missing parent (i.e. just 1 exists) and first check that the txid matches?